cs/src/core/expressions/DeserializerTransform.cs

```diff
@@ -83,6 +83,22 @@ internal class DeserializerTransform<R>
         static readonly MethodInfo bufferBlockCopy =
             Reflection.MethodInfoOf((byte[] a) => Buffer.BlockCopy(a, default(int), a, default(int), default(int)));
 
+        static readonly HashSet<string> immutableListSetTypeNames = new HashSet<string>
```

Comment:
Minor: a comment explaining why these are represented as strings (to avoid an assembly dependency) would be helpful here.

Reply:
Added explanatory comment